# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1733863724 -3600
# Node ID 1343b291721969fac55a05fac1ebce03e071e412
# Parent  1e2c417fb3163deb2b558dc7edacb197928c0218
Bug 1468925 - Enable advance to next msg by space bar when message browser has focus. r=mkmelin

Since no workable solution has been found in more than a year, I suggest again that this simple approach be considered for the time being. It solves the problem and passes all current tests.  If side effects actually occur in the future, the one self-contained block of code can be easily removed.

Differential Revision: https://phabricator.services.mozilla.com/D228166

diff --git a/mail/base/content/aboutMessage.js b/mail/base/content/aboutMessage.js
--- a/mail/base/content/aboutMessage.js
+++ b/mail/base/content/aboutMessage.js
@@ -355,8 +355,23 @@ window.addEventListener("DOMContentLoade
         Ci.nsIWebProgress.NOTIFY_ALL
       );
   }
 
+  if (Services.prefs.getBoolPref("mail.advance_on_spacebar")) {
+    getMessagePaneBrowser().addEventListener("keydown", ev => {
+      if (
+        ev.key == " " &&
+        !ev.altKey &&
+        !ev.ctrlKey &&
+        !ev.metaKey &&
+        ev.target.localName == "body"
+      ) {
+        ev.preventDefault();
+        top.goDoCommand("cmd_space", ev);
+      }
+    });
+  }
+
   window.dispatchEvent(
     new CustomEvent("aboutMessageLoaded", { bubbles: true })
   );
 });
